src/foam/classloader/NodeModelFileDAO.js

```diff
@@ -21,36 +21,51 @@ foam.CLASS({
   extends: 'foam.dao.AbstractDAO',
 
   properties: [
-    {
-      name: 'classpath'
-    }
+    { class: 'String', name: 'classpath' },
+
+    { name: 'sep', factory: function() { return require('path').sep; } },
+    { name: 'fs',  factory: function() { return require('fs');       } },
+    { name: 'vm',  factory: function() { return require('vm');       } }
   ],
 
   methods: [
     function find(id) {
       var foamCLASS = foam.CLASS;
       var self = this;
-      var model;
+      var model = null;
```

Comment:
The DAO should `resolve(null)` if the required code doesn't actually provide the requested class model.